# HG changeset patch
# User Rik <rik@octave.org>
# Date 1475372739 25200
#      Sat Oct 01 18:45:39 2016 -0700
# Branch stable
# Node ID 956f5c71dc3c7c1a7369d19258bfe52c4a84dacc
# Parent  fc9cd56986af075eea9a30d0689df1b19b728a08
Fix printing when axes has color "none" (bug #49243).

* print.m: Check whether list of top-level axes objects that need to change
color to white for the inverthardcopy option is empty before using said list.

diff --git a/scripts/plot/util/print.m b/scripts/plot/util/print.m
--- a/scripts/plot/util/print.m
+++ b/scripts/plot/util/print.m
@@ -395,24 +395,26 @@ function print (varargin)
       nfig = m + 2;
     endif
 
     if (do_hardcopy)
       ## Set background to white for all top-level axes objects
       hax = findall (opts.figure, "-depth", 1, "type", "axes",
                                   "-not", "tag", "legend",
                                   "-not", "color", "none");
-      m = numel (props);
-      for n = 1:numel(hax)
-        props(m+n).h = hax(n);
-        props(m+n).name = "color";
-        props(m+n).value{1} = get(hax(n), "color");
-        set (hax(n), "color", "white");
-      endfor
-      nfig += n;
+      if (! isempty (hax))
+        m = numel (props);
+        for n = 1:numel (hax)
+          props(m+n).h = hax(n);
+          props(m+n).name = "color";
+          props(m+n).value{1} = get(hax(n), "color");
+          set (hax(n), "color", "white");
+        endfor
+        nfig += n;
+      endif
     endif
 
     if (opts.force_solid != 0)
       h = findall (opts.figure, "-property", "linestyle");
       m = numel (props);
       for n = 1:numel (h)
         props(m+n).h = h(n);
         props(m+n).name = "linestyle";
